regnety/utils/tfrecords_utils.py

```diff
@@ -0,0 +1,184 @@
+"""Script to make and save TFRecords from ImageNet files"""
+
+import tensorflow as tf
+import os
+import random
+import json
+import argparse
+import math
+from .image_utils import *
+
+
+def _bytes_feature(value):
+    """Returns a bytes_list from a string / byte."""
+    if isinstance(value, type(tf.constant(0))):
+        value = (
+            value
+        )  # BytesList won't unpack a string from an EagerTensor.
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
+
+
+def _float_feature(value):
+    """Returns a float_list from a float / double."""
+    return tf.train.Feature(float_list=tf.train.FloatList(value=[value]))
+
+
+def _int64_feature(value):
+    """Returns an int64_list from a bool / enum / int / uint."""
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+
+# make_tfrecs -> for(make_single_tfrecord)-> make_dataset -> make_example
+def get_synset_labels(filepath):
+
+    with open(filepath, "r") as f:
+        raw_labels_dict = json.load(f)
+    labels_dict = dict()
+
+    for i in raw_labels_dict:
+        labels_dict[raw_labels_dict[i]["id"]] = (
+            int(i),
+            raw_labels_dict[i]["label"],
+        )
+    return labels_dict
+
+
+def get_files(data_dir, synset_filepath, shuffle = True):
+    all_images = tf.io.gfile.glob(os.path.join(data_dir, "*", "*.JPEG"))
+    all_synsets = [os.path.basename(os.path.dirname(f)) for f in all_images]
+
+    print(len(all_images))
+    all_indexes = list(range(len(all_images)))
+    if shuffle:
+        random.shuffle(all_indexes)
+
+    all_images = [all_images[i] for i in all_indexes]
+    all_synsets = [all_synsets[i][1:] + "-n" for i in all_indexes]
+
+    labels_dict = get_synset_labels(synset_filepath)
+
+    all_labels_int = [labels_dict[i][0] for i in all_synsets]
+
+    return all_images, all_labels_int, all_synsets
+
+
+def make_image(filepath):
+
+    image_str = tf.io.read_file(filepath)
+    
+
+    if is_png(filepath):
+        image_str = png_to_jpeg(image_str)
+
+    if is_cmyk(filepath):
+        image_str = cmyk_to_rgb(image_str)
+
+    image_tensor = tf.io.decode_jpeg(image_str)
+
+    if not is_rgb(image_tensor):
+        image_tensor = tf.image.grayscale_to_rgb(image_tensor)
+        
+    image_tensor = tf.cast(tf.image.resize(image_tensor, (512,512)), tf.uint8)
+
+    image_str = tf.io.encode_jpeg(image_tensor)
+
+    assert len(image_tensor.shape) == 3
+
+    return image_str
+
+
+def make_example(image_str, filepath, label, synset):
+
+    try:
+        example = tf.train.Example(
+            features=tf.train.Features(
+                feature={
+                    "image": _bytes_feature(image_str),
+                    "filename": _bytes_feature(
+                        os.path.basename(filepath)
+                    ),
+                    "label": _int64_feature(label),
+                    "synset": _bytes_feature(synset),
+                }
+            )
+        )
+    except:
```

Comment:
Why is this try except block here? The code is exactly the same in each. What problem does this solve?

Reply:
I'll remove it. It was needed when the script was meant to be ran in Eager mode.